src/arch/wasm/CodeGen.zig

```diff
@@ -2012,9 +2085,13 @@ fn emitUndefined(self: *Self, ty: Type) InnerError!void {
         },
         // As arrays point to linear memory, we cannot use 0xaaaaaaaa as the wasm
         // validator will not accept it due to out-of-bounds memory access);
-        .Array => try self.addImm32(@bitCast(i32, @as(u32, 0xaa))),
+        .Array => {
```

Comment:
Does it make sense to combine both cases?

Reply:
Good call. Also went ahead and made sure it writes 0xaa to its memory.